sql/connect/client/jdbc/src/main/scala/org/apache/spark/sql/connect/client/jdbc/SparkConnectResultSet.scala

```diff
@@ -147,8 +148,22 @@ class SparkConnectResultSet(
     getColumnValue(columnIndex, null: Date) { idx => currentRow.getDate(idx) }
   }
 
-  override def getTime(columnIndex: Int): Time =
-    throw new SQLFeatureNotSupportedException
+  override def getTime(columnIndex: Int): Time = {
+    getColumnValue(columnIndex, null: Time) { idx =>
+      val localTime = currentRow.get(idx).asInstanceOf[LocalTime]
+      // Convert LocalTime to milliseconds since midnight to preserve fractional seconds.
+      // Note: java.sql.Time can only store up to millisecond precision (3 digits).
+      // For TIME types with higher precision (TIME(4-9)), microseconds/nanoseconds are truncated.
+      // If user needs full precision,
+      // should use: getObject(columnIndex, classOf[java.time.LocalTime])
+      val millisSinceMidnight =
```

Comment:
Can we use `val millisSinceMidnight = java.time.temporal.ChronoUnit.MILLIS.between(LocalTime.MIDNIGHT, localTime)`?

Reply:
Yes, done